### PR TITLE
feat: include component and wrapper in return type for vue mount adapter

### LINKED ISSUE #24342: cypress/vue should return object containing vueWrapper and component
## Current Behavior
Currently vue mount returns the vue wrapper object.

## Desired Behavior
vue mount should return an object with the wrapper and component

```js
{
  wrapper: VueWrapper
  component: VueWrapper.vm
}
```

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://github.com/cypress-io/cypress/issues/24342

### User facing changelog

Vue Mount adapter now returns an object with both the `VueWrapper` (`wrapper`) and component instance (`component`).

### Additional details

Part of CT GA.

### Steps to test

You could do `const foo = cy.mount(SomeVueComponent)` then observe both `wrapper` and `component` properties are available on the returned value. `wrapper` contains all of [these methods](https://test-utils.vuejs.org/api/#wrapper-methods) from Vue Test Utils. Component contains various Vue related things.

### How has the user experience changed?

You can access the component instance, which is more in line with what other mount adapters do.

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)?  NOT YET
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/vue2/src/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /// <reference types="cypress" />
-import Vue from 'vue'
+import Vue, { Component } from 'vue'
```
Comment (ZachJW34) [inline npm/vue2/src/index.ts]:
```suggestion
import Vue from 'vue'
```

Unused

File: system-tests/project-fixtures/vue2-cli/src/components/HelloWorld.cy.js [hunk 2]
```diff
@@ -9,6 +9,15 @@ describe('<Logo />', () => {
       propsData: {
         msg: slotContent,
       },
+      data () {
```
Comment (ZachJW34) [inline system-tests/project-fixtures/vue2-cli/src/components/HelloWorld.cy.js:12]: Can we get a test for this with `vue3`

File: npm/vue/src/index.ts [hunk 3]
```diff
@@ -377,7 +430,10 @@ export function mount (componentOptions: any, options: any = {}) {
 
       // by returning undefined we keep the previous subject
       // which is the mounted component
-      return undefined
+      return {
```
Comment (ZachJW34) [inline npm/vue/src/index.ts:432]: We can remove the `cy.wrap(wrapper, { log: false })` above  as it's redundant now

**Review by ZachJW34 — APPROVED** [review]
Couple small comments but looks and works great, will approve but would like to see comments addressed before merge

File: system-tests/project-fixtures/vue2-cli/src/components/HelloWorld.cy.js (hunk 2 above)
Comment (lmiller1990) [reply system-tests/project-fixtures/vue2-cli/src/components/HelloWorld.cy.js:12]: We already have one here https://github.com/cypress-io/cypress/pull/24479/files#diff-f9a3bc77ab97b5f10f220de6ab30201be456aad25cb85e8197c52b2b90959684R24-R30

File: npm/vue/src/index.ts (hunk 3 above)
Comment (lmiller1990) [reply npm/vue/src/index.ts:432]: done

File: npm/vue2/src/index.ts (hunk 1 above)
Comment (lmiller1990) [reply npm/vue2/src/index.ts]: done

**Comment by lmiller1990** [comment]
Webkit system tests 🤷‍♂️ 

Everything else looks 💯

File: system-tests/project-fixtures/vue2-cli/src/components/HelloWorld.cy.js (hunk 2 above)
Comment (ZachJW34) [reply system-tests/project-fixtures/vue2-cli/src/components/HelloWorld.cy.js:12]: Sorry for missing that!!

File: npm/vue/src/index.ts [hunk 4]
```diff
@@ -377,7 +429,10 @@ export function mount (componentOptions: any, options: any = {}) {
 
       // by returning undefined we keep the previous subject
       // which is the mounted component
```
Comment (warrensplayer) [inline npm/vue/src/index.ts:431]: Comment no longer applicable

**Review by marktnoonan — APPROVED** [review]
Works great, chaining off the mount to get the wrapper works nice, putting the component instance in an alias and grabbing it there works nicely too. Probably no need for users to rely on `Cypress.vueWrapper` any more which will be sweet.

File: npm/vue/src/index.ts [hunk 5]
```diff
@@ -129,7 +129,10 @@ export function mount<V extends {}>(
     __vccOpts: any
   },
   options?: MountingOptions<any> & Record<string, any>
-): Cypress.Chainable<VueWrapper<ComponentPublicInstance<V>>>
+): Cypress.Chainable<{
+  wrapper: VueWrapper<ComponentPublicInstance<V>>
+  component: VueWrapper<ComponentPublicInstance<V>>['vm']
+}>
```
Comment (warrensplayer) [inline npm/vue/src/index.ts:135]: Still working on my Typescript, but would it make sense to introduce a generic return type like:
```
type MountReturn<C> = {
  wrapper: VueWrapper<C>
  component: C
}
```
to use for the return of all the mount overloads?

_Result_
```
export function mount<V extends {}>(
  originalComponent: {
    new (...args: any[]): V
    __vccOpts: any
  },
  options?: MountingOptions<any> & Record<string, any>
): Cypress.Chainable<MountReturn<V>>
```

Would that work for all the overload cases?

**Review by warrensplayer — COMMENTED** [review]
This PR already got merged while I was in the middle of reviewing, but will drop these comments here for future consideration.

**Pull request merged**